### DISCUSSION TITLE
Active Storage media library / file manager

### DISCUSSION THREAD

**Discussion by sedubois** [avo-hq/avo#2073] [Feature requests]
I understand the [files](https://docs.avohq.io/3.0/fields/files.html) field allows to upload new files but not to reuse or otherwise manage the ones already in Active Storage.

An Active Storage file manager / gallery / picker would be needed to browse, organize, upload, delete active storage blobs and variants. For instance it would lead to duplication if the same file has to be uploaded every time (when the same image/file needs to be attached to different records). For instance, when there is an article in several languages, the text in the Action Text editor needs to differ but often the embedded images should remain the same and be reused across the different translations. One may also simply need a way to visually browse/search all Active Storage blobs to be aware of what's there, find orphan records, merge duplicates, view/edit metadata, see related variants, etc.

There are three places where it seems necessary to offer access to this Active Storage file manager: 
- for the files field
- in the Trix editor, for adding embeds
- direct access from the navigation

**Comment by sedubois** [comment]
There is a presentation [coming up](https://rubyonrails.org/world/2024/session/andrea-fomera) at Rails World on this subject.

> You’ll learn how to build ... a rich media library that allows users to see, modify and upload files they’ve uploaded that can be selected from in forms in place of uploading a new file

**Reply by adrianthedev** [reply]
Yes, this is interesting!
I do see the advantage of having something like that in Avo.

**Reply by adrianthedev** [reply]
I'll go see that talk. Are you coming to RW?

**Reply by sedubois** [reply]
I won't be coming unfortunately but looking forward to watching the talks!

**Reply by sedubois** [reply]
Here's the talk & code, Media Library part is @ 18 min:
https://www.youtube.com/watch?v=r4X8e1ARUj8
https://github.com/afomera/rails-world-2024
https://afomera.dev/speaking

**Reply by adrianthedev** [reply]
We started to work on this in https://github.com/avo-hq/avo/pull/3561

**Comment by Paul-Bob** [comment]
https://docs.avohq.io/3.0/media-library.html
